### DISCUSSION TITLE
How to run this on Podman?

### DISCUSSION THREAD

**Discussion by vordenken** [wg-easy/wg-easy#244] [Q&A]
I would like to run this with podman on fedora iot but so far its not working.
Anybody runs it like that and could help?

**Comment by eandersons** [comment]
What exactly did you try and what was the outcome? What command did you use to create a WireGuard Easy container? Did WireGuard Easy container fail to start? Was container started successfully but client(s) cannot connect to VPN?

Without additional information nobody can guess what and why something is not working. The statement "it is not working" does not help to help at all.

**Reply by vordenken** [reply]
Hey, thanks for the reply. Here are the container logs: 
```
2022-04-12T12:04:53.458ZServer Listening on http://0.0.0.0:51821
2022-04-12T12:04:53.459Z WireGuard Loading configuration...
$ wg genkey
$ echo ***hidden*** | wg pubkey
2022-04-12T12:04:53.477Z WireGuard Configuration generated.
2022-04-12T12:04:53.477Z WireGuard Config saving...
[Error: EACCES: permission denied, open '/etc/wireguard/wg0.json'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'open',
  path: '/etc/wireguard/wg0.json'
}
```

I try to start the container with this command: (Its basically the same as in the readme but with _podman_ instead of _docker_ : 
```
podman run -d \
  --name=wg-easy \
  -e WG_HOST=🚨testdomain.com \
  -e PASSWORD=🚨supersafepw \
  -v ~/.wg-easy:/etc/wireguard \
  -p 51820:51820/udp \
  -p 51821:51821/tcp \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
  --restart unless-stopped \
  weejewel/wg-easy
```

From the container logs it seems that the container has no permission to read/write files but its running as my user and so has rights to its home directory. The same problem occurs if I run it with _sudo_ ...


If you need additional info let me know :)

**Reply by eandersons** [reply]
I could not reproduce the same error but it seems to be related to Node.

What is the output of the following command: `podman run --entrypoint ash --volume ~/.wg-easy:/etc/wireguard  weejewel/wg-easy -c 'whoami && ls -ahl /etc/wireguard'`?

Side note: multiline code blocks can be formatted by [wrapping them between triple backticks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks):

````markdown
Some content.

```
multiple
lines
with
monospace
font
```

Content continues.
````

**Reply by vordenken** [reply]
Here's the output: 
```
root
total 0
ls: can't open '/etc/wireguard': Permission denied
```

Thanks for the tip, I knew this didn't look right :) Updated the old comment.

**Reply by eandersons** [reply]
It starts to seem that this is not related to Node after all.
What is the output of the following command: `podman run --entrypoint ash --volume ~/.wg-easy:/etc/wireguard weejewel/wg-easy -c 'ls -ahl /etc | grep wireguard' && id -u`?

By the way, what operating system are you using?

**Reply by vordenken** [reply]
Here's the output:
```
drwxr-xr-x    2 root     root        4.0K Apr 11 10:18 wireguard
1000
```
Im using fedora iot (version 35).
Like I said it seems like a permission problem. But I'm running podman with my user and the "volume" is my home folder? 
Btw if I create a real volume and start the container with it, it works.

**Reply by eandersons** [reply]
Yes, it is a permissions related problem, but I am not sure why it happens.

A bad suggestion would be to do `chmod -R 777 ~/.wg-easy`.
Maybe a better suggestion would be `chgrp -R root ~/.wg-easy && chmod -R g+w ~/.wg-easy` (or `chown -R root ~/.wg-easy && chmod -R g+w ~/.wg-easy`) but I am not sure if will solve the issue.
Or you could just use a Docker volume to store the WireGuard configuration.

**Comment by pheiduck** [comment]
Head over to the wiki page: https://github.com/wg-easy/wg-easy/wiki/Using-WireGuard-Easy-with-Podman
